### PR TITLE
Filter validation list by a list of access codes (od_mtl #68)

### PR DESCRIPTION
**Pull request by kaligrafy**
## Summary

Lets validators filter the admin validation list by **several access codes at once**, instead of searching codes one by one. Implements od_mtl [#68](https://github.com/chairemobilite/od_mtl/issues/68).

- The access code filter accepts several codes, either **typed** (separated by commas or new lines) or **imported from a CSV file** (one code per line).
- A single code keeps the existing exact-match behavior; several codes match any interview having one of them.
- CSV values are validated with the existing `accessCodeValidation`, which also discards a header row (e.g. `accessCode`).

## Changes

**Backend** (commit 1)
- New `in` operator for the interview list query filters. It expands to a SQL `IN (?, ?, ...)` clause on `response.*` fields (parameterized, no SQL injection). An empty list matches nothing. Array values with other operators keep their previous AND semantics.

**Frontend** (commit 2)
- `InterviewByCodeFilter`: the input becomes a small textarea; codes can be comma/newline separated. Added a CSV import button (the typed field is not filled on import, only a "N codes imported" counter is shown, to avoid a huge field).
- New `parseAccessCodesFromCsv` util (one code per line, first column only, trims quotes/whitespace, dedupes, validates format).
- i18n keys (fr/en) and CHANGELOG entry.

## Notes

- Access code format validation reuses `accessCodeValidation` (evolution-common), i.e. the default 8-digit format. Projects using a different format would need a parameterizable validator (out of scope here).
- Known, **separate pre-existing issue**: the validation list reads/filters `response.accessCode`, not `corrected_response.accessCode`, so a validator-corrected access code is not reflected in the list. Tracked separately, not addressed here.

## Test plan

- [x] `parseAccessCodesFromCsv` parametric unit tests (header dropped, invalid codes dropped, dedupe, multi-column, CRLF, quotes)
- [x] `interviews.db` `in` operator tests (both codes, partial, none, empty list)
- [x] Type-check (backend + frontend)
- [x] Manual: type `1111-1111,2222-2222`, and import a CSV with a header row, confirm the list filters correctly

## Summary by CodeRabbit

* **New Features**
  * Added support for filtering interviews by one or multiple access codes, entered as comma/newline-separated text or imported from a CSV/plain-text file (one code per line).
  * Updated admin filter behavior to match any provided code, with UI feedback for imported counts and import errors.

* **Tests**
  * Added backend and frontend test coverage for multi-code “in” filtering and robust CSV parsing (including edge cases and validation).

* **Documentation / Localization**
  * Expanded admin UI translations for the new filter input, import controls, and related messages (including pluralization and error text).

### CODE REVIEW

File: packages/evolution-backend/src/models/interviews.db.queries.ts [hunk 1]
```diff
@@ -280,7 +280,7 @@ const update = async (
 /**
  * Array of valid operator keys for ValueFilterType
  */
-export const VALID_OPERATORS = ['eq', 'gt', 'lt', 'gte', 'lte', 'not', 'like'] as const;
+export const VALID_OPERATORS = ['eq', 'gt', 'lt', 'gte', 'lte', 'not', 'like', 'in'] as const;
```
[the remark this answers is not in the document]
Comment (kaligrafy) [reply packages/evolution-backend/src/models/interviews.db.queries.ts:283]: fixed

File: packages/evolution-frontend/src/components/admin/validations/InterviewByCodeFilter.tsx [hunk 2]
```diff
@@ -27,22 +32,78 @@ export const InterviewByCodeFilter = ({
     const [currentValue, setCurrentValue] = React.useState(
         !_isBlank(filterValue) && filterValue.value ? filterValue.value : filterValue
     );
```
[the remark this answers is not in the document]
Comment (kaligrafy) [reply packages/evolution-frontend/src/components/admin/validations/InterviewByCodeFilter.tsx:39]: fixed

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+- **Validation list: filter by a list of access codes**: In the admin validation list, the access code filter now accepts a CSV file (one access code per line). Imported codes filter the list to interviews matching any of them, instead of searching codes one by one (od_mtl [#68](https://github.com/chairemobilite/od_mtl/issues/68)). Internally, this adds an `in` operator to the interview list query filters.
```
Comment (tahini) [inline CHANGELOG.md]: Ne pas mettre un issue dasn od_mtl comme issue à fixer. Il faut plutôt ouvrir une issue ici qui explique le problème générique (hors od_mtl) et pointer l'issue d'od-mtl vers cet issue dans Evolution (comme ça on track séparément le problème à régler de la mise à jour d'Evolution correspondant).

aussi note changelog est beaucoup trop verbose. À part pour les breaking changes qu'on peut détailler un peu, faut être minimal pour chaque ligne et pointer plutôt vers l'issue fixé/commit pour plus d'information.

**Review by tahini — COMMENTED** [review]
Juste faire une issue dans Evolution et la fixer via ce commit. Sinon, ça m,a l,air ben beau

File: CHANGELOG.md (hunk 3 above)
Comment (kaligrafy) [reply CHANGELOG.md]: #1660

Comment (kaligrafy) [reply CHANGELOG.md]: simplifié le changelog aussi

File: packages/evolution-frontend/src/components/admin/validations/InterviewByCodeFilter.tsx [hunk 4]
```diff
@@ -24,25 +30,101 @@ export const InterviewByCodeFilter = ({
...
+                label=""
+                size="small"
+                title={t('admin:interviewByCodeFilter:importButton')}
+            />
+            {importedCodeCount !== undefined && (
+                <span style={{ marginLeft: '0.5rem' }}>
+                    {t('admin:interviewByCodeFilter:importedCount', { count: importedCodeCount })}
+                </span>
+            )}
+            {importError && <FormErrors errors={['admin:interviewByCodeFilter:importError']} />}
```
[the remark this answers is not in the document]
Comment (kaligrafy) [reply packages/evolution-frontend/src/components/admin/validations/InterviewByCodeFilter.tsx:127]: @tahini ça se fait pas tout seul?

[the remark this answers is not in the document]
Comment (tahini) [reply packages/evolution-frontend/src/components/admin/validations/InterviewByCodeFilter.tsx:127]: @kaligrafy oui, ça se fait tout seul, FormErrors reçoit des props de type `TranslatableMessage`, donc on ne s'attend pas à ce qu'ils soient déjà traduits

[the remark this answers is not in the document]
Comment (kaligrafy) [reply packages/evolution-frontend/src/components/admin/validations/InterviewByCodeFilter.tsx:127]: parfait!

**Review by tahini — APPROVED** [review]
